src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,40 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Could not save phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> criteriaQuery = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = criteriaQuery.from(Phone.class);
+            List<Predicate> predicates = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                predicates.add(root.get(entry.getKey()).in((Object[]) entry.getValue()));
```

Comment:
```suggestion
                predicates.add(root.get(entry.getKey()).in(entry.getValue()));
```
do we need this cast?